### PR TITLE
ENH Add support for asynchronous iterators to JsProxy

### PR DESCRIPTION
**Pull request by hoodmane**
In JavaScript there is no way to tell the difference between a async generator and a sync generator without trying to iterate them, so we just support `anext` and `next` on both and raise an error if it turns out you used the wrong one.

On top of #3294.

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by hoodmane** [comment]
I ended up pushing a bunch of different stuff to this one branch, it could probably use to be split up.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane! I have some minor comments about docstrings, otherwise LGTM.

File: src/py/pyodide/ffi/__init__.py [hunk 1]
```diff
@@ -28,6 +30,8 @@
     "JsDoubleProxy",
     "JsGenerator",
     "JsIterator",
+    "JsIterable",
```
[the remark this answers is not in the document]
Comment (ryanking13) [reply src/py/pyodide/ffi/__init__.py:33]: Oh, okay this was fixed in #3297.

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13!

**Pull request merged**